File: tests/core/tools/codac2_tests_Approx.py
#!/usr/bin/env python

#  Codac tests
# ----------------------------------------------------------------------------
#  \date       2024
#  \author     Simon Rohou
#  \copyright  Copyright 2024 Codac Team
#  \license    GNU Lesser General Public License (LGPL)

import unittest
from codac import *
import sys

class TestApprox(unittest.TestCase):

  def tests_approx(self):

    self.assertFalse(Interval(0.1) == Interval(1.)/Interval(10.))
    self.assertTrue(Approx(Interval(0.1)) == Interval(1.)/Interval(10.))
    self.assertTrue(Interval(0.1) == Approx(Interval(1.)/Interval(10.)))

    #self.assertTrue(IntervalVector([[0.1]]) != IntervalVector([1.])/Interval(10.))
    #self.assertTrue(Approx(IntervalVector([[0.1]])) == IntervalVector([1.])/Interval(10.))
    #self.assertTrue(IntervalVector([[0.1]]) == Approx(IntervalVector([1.])/Interval(10.)))
      
if __name__ ==  '__main__':
  unittest.main()

File: tests/core/tools/codac2_tests_Approx.cpp
/** 
 *  Codac tests
 * ----------------------------------------------------------------------------
 *  \date       2024
 *  \author     Simon Rohou
 *  \copyright  Copyright 2024 Codac Team
 *  \license    GNU Lesser General Public License (LGPL)
 */

#include <catch2/catch_test_macros.hpp>
#include <codac2_Interval.h>
#include <codac2_IntervalVector.h>
#include <codac2_Approx.h>
#include <codac2_arithmetic.h>

using namespace std;
using namespace codac2;

TEST_CASE("Approx")
{
  CHECK(Interval(0.1) != Interval(1.)/Interval(10.));
  CHECK(Approx(Interval(0.1)) == Interval(1.)/Interval(10.));
  CHECK(Interval(0.1) == Approx(Interval(1.)/Interval(10.)));

  CHECK(IntervalVector({{0.1}}) != IntervalVector({{1.}})/Interval(10.));
  CHECK(Approx(IntervalVector({{0.1}})) == IntervalVector({{1.}})/Interval(10.));
  CHECK(IntervalVector({{0.1}}) == Approx(IntervalVector({{1.}})/Interval(10.)));
}